src/deeponet.jl

```diff
@@ -90,43 +91,69 @@ julia> branch_net = Chain(Dense(64 => 32), Dense(32 => 32), Dense(32 => 16));
 julia> trunk_net = Chain(Dense(1 => 8), Dense(8 => 8), Dense(8 => 16));
 
 julia> deeponet = DeepONet(branch_net, trunk_net)
-@compact(
-    branch = Chain(
+Branch net :
+(
+    Chain(
         layer_1 = Dense(64 => 32),      # 2_080 parameters
         layer_2 = Dense(32 => 32),      # 1_056 parameters
         layer_3 = Dense(32 => 16),      # 528 parameters
     ),
-    trunk = Chain(
+)
+
+Trunk net :
+(
+    Chain(
         layer_1 = Dense(1 => 8),        # 16 parameters
         layer_2 = Dense(8 => 8),        # 72 parameters
         layer_3 = Dense(8 => 16),       # 144 parameters
     ),
-) do (u, y)
-    t = trunk(y)
-    b = branch(u)
-    @argcheck ndims(t) == ndims(b) + 1 || ndims(t) == ndims(b)
-    @argcheck size(t, 1) == size(b, 1) "Branch and Trunk net must share the same amount of nodes in the last layer. Otherwise Σᵢ bᵢⱼ tᵢₖ won't work."
-    b_ = if ndims(t) == ndims(b)
-            b
-        else
-            reshape(b, size(b, 1), 1, (size(b))[2:end]...)
-        end
-    return dropdims(sum(t .* b_; dims = 1); dims = 1)
-end       # Total: 3_896 parameters,
-          #        plus 0 states.
+)
+
+julia> branch_net = Chain(Dense(64 => 32), Dense(32 => 32), Dense(32 => 16));
+
+julia> trunk_net = Chain(Dense(1 => 8), Dense(8 => 8), Dense(8 => 16));
+
+julia> additional = Chain(Dense(1 => 4));
```

Comment:
input for  additional  layer should be size of inner embedding size

Reply:
So it not need `reduction/sum/dropdim` before `additional layer.` It should be `additional = Chain(Dense(16 => 4));` here. Otherwise It's created a bottleneck and we lose information here.

Reply:
Should be fixed now. Using the `linear` layer as `additional` layer for the cases where we do not have the `additional` layer did not seem ideal to me because it would imply weighted sum, where the weights would be learnt during training, but since DeepONets by default take the dot product, aka non-weighted sum, which could be required by many users.